src/tools/kdb/cmerge.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "cmerge.hpp"
+#include "kdbmerge.h"
+#include "keyset.hpp"
+#include <cmdline.hpp>
+#include <iostream>
+#include <keysetio.hpp>
+#include <string>
+
+
+CMergeCommand::CMergeCommand ()
+{
+}
+
+CMergeCommand::~CMergeCommand ()
+{
+}
+
+int CMergeCommand::execute (Cmdline const & cl ELEKTRA_UNUSED)
+{
+	if (cl.arguments.size () < 4)
+	{
+		throw invalid_argument ("Wrong number of arguments! At least 4 arguments needed");
+	}
+	kdb::Key oursRoot = cl.createKey (0);
+	kdb::Key theirsRoot = cl.createKey (1);
+	kdb::Key baseRoot = cl.createKey (2);
+	kdb::Key resultRoot = cl.createKey (3);
+	int strategy = MERGE_STRATEGY_ABORT;
+	if (cl.strategy == "preserve")
+	{
+		/** This is here for compatibility. The old merge has preserve as default as defined in cmdline.cpp.
```

Comment:
Here the user will not read it, add to man page. Or actually compatibility is not really important in your case.

Reply:
The user doesn't have to read it. `preserve` is the default for a command line option `strategy` in cmdline.cpp. I didn't want to make something like `cstrategy` so I decided to work around by catching it here. This compatibility will not be required anymore as soon as we replace merge with cmerge.